Fix sync script

bullet-tooth opened the pull request:
Now it catches file movements and renames.

dmitry-timofeev commented:
> because the core contains redundant proto files (tests, doc_tests).

Please specify the reasoning for explicit listing in the script documentation.

bullet-tooth commented:
@dmitry-timofeev @vitvakatu the following improvements are done:
- now it pick-ups new files as well as renames and deletions
- added description and example how to use sync feature within git submodules
- added file exclusions i.e. known files to ignore

please review

vitvakatu commented on README.md:
```diff
@@ -1,2 +1,21 @@
+git fetch
+git checkout master
+./proto-sync.sh 228d861544a93a65961e021cb6838c8180675861 # specify desired revision id
+# It will ask you to review all changes and then pushes them to the repote
```
> them to the repote

To the remote or to the repo? :)

vitvakatu commented on README.md:
```diff
@@ -1,2 +1,21 @@
+./proto-sync.sh 228d861544a93a65961e021cb6838c8180675861 # specify desired revision id
+# It will ask you to review all changes and then pushes them to the remote
+cd .. # go to main repo
+git commit -am "sync complete" # now you reflacted submodule changes in your branch
```
```suggestion
git commit -am "sync complete" # now you reflected submodule changes in your branch
```
